client/extensions/woocommerce/state/data-layer/products/index.js

```diff
@@ -75,3 +84,46 @@ export function handleProductRequest( { dispatch }, action ) {
 	const updatedSuccessAction = updatedAction( siteId, action, successAction );
 	dispatch( get( siteId, 'products/' + productId, updatedSuccessAction, failureAction ) );
 }
+
+export function handleProductsRequestSuccess( store, action, next, response ) {
+	const { siteId, page } = action;
+	const { headers, body, status } = response.data;
+
+	// We needed to envelope the reponse to get the number of products and pages. see handleProductsRequest
+	if ( status !== 200 ) {
```

Comment:
I can make the comment above it a bit clearer, but we have to use https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_envelope to get back the number of products and pagination pages from the API. The headers that get returned to the data layer are from `public-api.wordpress.com`/ the proxy.